doc/refman/RefMan-ssr.tex

```diff
@@ -0,0 +1,4671 @@
+\dots\\
+F_k\ \ssrC{:}\ P_k \\
+\dots \\[3pt]
+\hline\hline\\[-8pt]
+\ssrC{forall}\ \ssrC{(}x_\ell\ \ssrC{:}\ T_\ell\ssrC{)}\ \dots,\\
+\ssrC{let}\ y_m\ \ssrC{:=}\ b_m\ \ssrC{in}\ \dots\ \ssrC{in}\\
+P_n\ \ssrC{->}\ \dots\ \ssrC{->}\ C
+\end{array}\end{displaymath}
+The \emph{goal} to be proved appears below the double line; above the line is
+the \emph{context} of the sequent, a set of declarations of
+\emph{constants}~$c_i$, \emph{defined constants}~$d_i$, and
+\emph{facts}~$F_k$ that can be used to prove the goal (usually, $T_i,
+T_j\;:\;\ssrC{Type}$ and $P_k\;:\;\ssrC{Prop}$). The various kinds of
+declarations can come in any order. The top part of the context
+consists of declarations produced by the \ssrC{Section} commands
+\ssrC{Variable}, \ssrC{Let}, and \ssrC{Hypothesis}. This \emph{section context}
+is never affected by the \ssr{} tactics: they only operate on the
+the lower part --- the \emph{proof context}.
+As in the figure above, the goal often decomposes into a series of
+(universally) quantified \emph{variables}
+$\ssrC{(}x_\ell\;\ssrC{:}\;T_\ell\ssrC{)}$, local \emph{definitions}
+$\ssrC{let}\;y_m\:\ssrC{:=}\;b_m\;\ssrC{in}$, and \emph{assumptions}
+$P_n\;\ssrC{->}$, and a \emph{conclusion}~$C$ (as in the context, variables,
+definitions, and assumptions can appear in any order). The conclusion
+is what actually needs to be proved --- the rest of the goal can be
+seen as a part of the proof context that happens to be ``below the line''.
+
+However, although they are logically equivalent, there are fundamental
+differences between constants and facts on the one hand, and variables
+and assumptions on the others. Constants and facts are
+\emph{unordered}, but \emph{named} explicitly in the proof text;
+variables and assumptions are \emph{ordered}, but \emph{unnamed}: the
+display names of variables may change at any time because of
+$\alpha$-conversion.
+
+Similarly, basic deductive steps such as \ssrC{apply} can only operate on
+the goal because the Gallina terms that control their action (e.g.,
+the type of the lemma used by \ssrC{apply}) only provide unnamed bound
+variables.\footnote{Thus scripts that depend on bound variable names, e.g.,
+via \ssrC{intros} or \ssrC{with}, are inherently fragile.}  Since the proof
+script can only refer directly to the context, it must constantly
+shift declarations from the goal to the context and conversely in
+between deductive steps.
+
+In \ssr{} these moves are performed by two \emph{tacticals} `\ssrC{=>}'
+and `\ssrC{:}', so that the bookkeeping required by a deductive step can
+be directly associated to that step, and that tactics in an \ssr{}
+script correspond to actual logical steps in the proof rather than
+merely shuffle facts. Still, some isolated bookkeeping is unavoidable,
+such as naming variables and assumptions at the beginning of a proof.
+\ssr{} provides a specific \ssrC{move} tactic for this purpose.
+%; about one out of every six tactics is a \ssrC{move}.
```

Comment:
This was true before ssr 1.3 war released, it was 2011 or so... it should really be removed ;-)